### PR TITLE
#171380749 reset password via email

### PR DESCRIPTION
**Pull request by Ugizwenayo-Divine**
#### What does this PR do? 
reset a password via email

#### Description of Task to be completed?
- created an email controller
- created an email helper
- created email templates
- add all tests regarding resetting a password
- documented the task using swagger

##### Not in postman
In case you don't have this repository on your local computer
- clone this repo by `git clone https://github.com/Stackup-Rwanda/stackup2-barefoot-backend.git`
- `cd stackup2-barefoot` to go in the root directory of the project
- `git pull origin ft-reset-password-171380749` to pull this branch
- npm install or yarn add to install all the packages
- provide values to all those environment variable 
TEST_DB_USERNAME
TEST_DB_PASSWORD
TEST_DB_NAME
TEST_DB_HOSTNAME
TEST_DB_PORT
API_KEY
- make sure you are in a test environment
- npm run test

##### Testing in postman
- set all those environments variable
NODE_ENV to development
DEV_DB_USERNAME
DEV_DB_PASSWORD
DEV_DB_NAME
DEV_DB_HOSTNAME
DEV_DB_PORT
API_KEY
- run npm seed 
to enter the data in the database
- run npm start
- to send an email `http://localhost:3000/api/auth/resetpassword`
- to update a password `http://localhost:3000/api/auth/resetpassword/token`
#### Any background context you want to provide?
- you need to install node version 12
- you need to create an account on https://sendgrid.com/

#### What are the relevant pivotal tracker stories?
[#171380749](https://www.pivotaltracker.com/story/show/171380749)

#### Screenshots (if appropriate)
- The email to reset password 

Capture

- The after success password reset

Capture2

### CODE REVIEW

**Review by since9629 — CHANGES_REQUESTED** [review]
@Ugizwenayo-Divine could you resolve all conflicts this PR has with base branch

**Review by ricahcyuzuzo — COMMENTED** [review]
@Ugizwenayo-Divine Make sure you fix Travis CI Issues!

**Review by ricahcyuzuzo — APPROVED** [review]
Good Job!

File: tests/authentication.test.js [hunk 1]
```diff
@@ -136,4 +135,102 @@ describe('Login', () => {
...
+    const token = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MSwiZmlyc3ROYW1lIjoiRGl2aW5lIiwibGFzdE5hbWUiOiJ1Z2l6d2VuYXlvIiwiZW1haWwiOiJ1Z2l6d2VuYXlvZGlueUBnbWFpbC5jb20iLCJnZW5kZXIiOiJGZW1hbGUiLCJhZGRyZXNzIjoia2FtZW1iZSIsInJvbGUiOiJ1c2VyIiwiaXNWZXJpZmllZCI6ZmFsc2UsImNyZWF0ZWRBdCI6IjIwMjAtMDMtMTFUMDg6NTY6MTMuMDU3WiIsInVwZGF0ZWRBdCI6IjIwMjAtMDMtMTFUMDg6NTY6MTMuMDU3WiIsImlhdCI6MTU4MzkxNzEwMywiZXhwIjoxNTg0MDAzNTAzfQ.bwwiL9PwiEUtGxmmgCIAChMTph2-ZeoOfkuJi4fIpqk';
+    const pass = {
+      password: 'ugizwe'
+    };
+    chai
+      .request(server)
+      .post(`/api/auth/resetpassword/1/${token}`)
+      .send(pass)
+      .end((err, res) => {
+        res.should.have.status(200);
```
Comment (Muhire-Josue) [inline tests/authentication.test.js]: Could you use the status code in `src/utils/statusCode.js` like this `res.should.have.status(statusCodes.ok)`

File: tests/authentication.test.js [hunk 2]
```diff
@@ -136,4 +135,102 @@ describe('Login', () => {
...
+        res.body.message.should.be.equal('The password has been changed');
```
Comment (Muhire-Josue) [inline tests/authentication.test.js]: same thing for the message, you can do this `res.body.message.should.be.equal(messages.passwordChanged)`

**Review by Muhire-Josue — APPROVED** [review]
great work,

File: tests/authentication.test.js (hunk 1 above)
Comment (Ugizwenayo-Divine) [reply tests/authentication.test.js]: let me work on them

**Review by since9629 — APPROVED** [review]

**Review by the22mastermind — CHANGES_REQUESTED** [review]
Great work on this PR @Ugizwenayo-Divine . However I left you some comments.

File: public/api-docs/swagger.json [hunk 3]
```diff
@@ -106,6 +106,75 @@
...
       }
+    },
+    "/auth/user/:id":{
+      "post": {
+        "summary": "Reset email",
+        "produces": [
+          "application/json"
+        ],
+        "consumes":[
+          "application/x-www-form-urlencoded"
```
Comment (the22mastermind) [inline public/api-docs/swagger.json]: Consider changing changing this line to "application/json"

File: public/api-docs/swagger.json [hunk 4]
```diff
@@ -189,7 +274,8 @@
         }
       }
     }
-  },"login-definition":{
+  },
+  "login-definition":{
```
Comment (the22mastermind) [inline public/api-docs/swagger.json]: You could move all the responses inside this `login-definition` block into `responses` block

**Review by the22mastermind — APPROVED** [review]

**Review by the22mastermind — APPROVED** [review]

**Review by muchristian — APPROVED** [review]

**Review by mifeille — APPROVED** [review]

**Pull request merged**